main.rb

```diff
@@ -0,0 +1,8 @@
+require './app'
+
+def main
+  app = App.new
+  app.start
+end
+
+main
```

Comment:
These were the instructions you need to follow but in your case, it only runs the `app.rb` but the other instruction are not met. Moreover, the app quits automatically and not work as expected when done any functionality.

### INSTRUCTIONS:-


In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:

1- Present the user with a list of options to perform.
2- Lets users choose an option.
3- If needed, ask for parameters for the option.
4- Have a way to quit the app.


### VIDEO DEMONSTRATION OF APP QUITS UNEXPECTEDLY:-



https://user-images.githubusercontent.com/94223367/236638800-33afe927-7414-489c-b080-6f35c47e8198.mp4

### CODE DEMONSTRATION:-

You can have a code somewhat like that not exactly similar to meet the above requirement ✅ 

```
# main.rb
def main
  app = App.new()

  response = nil

  # *Present the user with a list of options to perform.
  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by enterin a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    puts "4 - Create a book"
    puts "5 - Create a rental"
    puts "6 - List all rentals for a given person id"
    puts "7 - Exit"
    # *Lets users choose an option.
    response = gets.chomp

    case response
    when "1"
      app.list_books()
    when "2"
      app.list_people()
    when "3"
      app.create_person()
    when "4"
      app.create_book()
    when "5"
      app.create_rental()
    when "6"
      app.list_rentals_for_person_id()
    when "7"
      # *Have a way to quit the app.
      puts "Thank you for using this app!"
    end

    puts "\n"
  end
end

# 'main' method should be invoked at the end of the main.rb file, hint in this lesson https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/console_apps.md#console-app-entry-point
main()


```

### MORE HELPFUL RESOURCES:-

[LINK](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/console_apps.md#console-app-entry-point)